library/x509_crt.c

```diff
@@ -2523,7 +2523,7 @@ static int x509_crt_verify_chain(
     int signature_is_good;
     unsigned self_cnt;
     mbedtls_x509_crt *cur_trust_ca = NULL;
-    mbedtls_x509_time now;
+    mbedtls_x509_time now = { 0 };
```

Comment:
```suggestion
    mbedtls_x509_time now;
    memset(&now, 0, sizeof(mbedtls_x509_time));
```

AIUI using = { 0 } will initialise only the first element of a struct.

Reply:
`= {0}` is perfectly correct C to initialize all the fields of a structure to logical zero. Clang's complaint is that it doesn't know whether you meant “initialize the first field to 0, and all the other fields to 0” or “initialize the first field to 0 [and I forgot to initialize the remaining fields]”. So to reassure `-Wmissing-field-initializers` we must initialize all the fields.

Modern versions (at least from 6.0 onwards, I don't know about 4 and 5) exclude the specific case of `{0}` from that warning, by the way. But we test with Clang 3.5 which does.

Reply:
Yes. It's interesting that old versions of clang complain about doing this for a struct, but not for an array - it feels like it's essentially a compiler bug in terms of warning about something that's actually a (correct) common idiom.